tt_metal/impl/dispatch/memcpy.hpp

```diff
@@ -68,6 +67,12 @@ static inline void memcpy_to_device(void *__restrict dst, const void *__restrict
                 src8 += sizeof(int32_t);
                 dst8 += sizeof(int32_t);
             }
+            n -= n / sizeof(int32_t) * sizeof(int32_t);
+            for (size_t i = 0; i < n; ++i) {
```

Comment:
is it memcpy?

Reply:
@pgkeller I recall you mentioning we'd always need to use non-temporal stores here and can't switch between them

Reply:
right.  if you do a store, the data will be pulled into cache and you then run the risk of a cacheline flush later corrupting data.  I think w/ the pcie device being cache coherent that that may solve the issue (device pulls from memory, flushing the cache) but I'm not sure if the flush is an eviction or just a write or if that behavior is defined or not.

ideally, these writes would take aligned buffers and we'd control the allocation to ensure they are always aligned, then the copies would always use si128 writes (even potentially writing off the end by a few bytes)